### PR TITLE
improved xlC simd pragmas 

### PR DESCRIPTION
**Pull request by artv3**
This pull request updates the simd macros used in the xlc compiler. Furthermore, I observed that cmake versions < 3.11 will not recognize the xlC compiler.  It is necessary to use cmake version >= 3.11 and use the following flag -qxlcompatmacros to define the macros. Lastly, I observed improved performance using the following flags "-O3 -qhot -qarch=pwr8 -qtune=pwr8"  (on the power8's) with the new simd macros.

### CODE REVIEW

**Comment by artv3** [comment]
@davidbeckingsale , could I get a review here?

**Comment by artv3** [comment]
@rhornung67 , @davidbeckingsale , I added some updates to the simd directive list here. I think this branch is good to go.

File: include/RAJA/config.hpp.in [hunk 1]
```diff
@@ -270,14 +270,13 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 
 #define RAJA_ALIGN_DATA(d) __alignx(RAJA::DATA_ALIGN, d)
 
-//#define RAJA_SIMD  _Pragma("simd_level(10)")
+
```
Comment (davidbeckingsale) [inline include/RAJA/config.hpp.in]: You can remove this spurious line.

**Comment by rhornung67** [comment]
@artv3 we should be capturing things like your comments at the top about compiler flags in the user guide. I will add an issue for this. Will you have time to create a new page in the configure and builds section? Something like "Compiler-specific build options"....

**Comment by artv3** [comment]
@rhornung67 , yes absolutely. I'll add content on the compiler-specific build options.

**Comment by artv3** [comment]
@rhornung67 , Should the page be a separate PR?

**Comment by rhornung67** [comment]
@artv3 make a new branch for doc updates. I have some things to add too. Plus, we can add a few other things we discussed at the JIRA issue planning meeting.

**Comment by artv3** [comment]
@davidbeckingsale , this branch got left behind. Can I get a re-approval?

**Pull request merged**